applications: serial_lte_modem: document CMUX and PPP (#14245)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/slm_description.rst

* :file:`overlay-full_fota.conf` - Configuration file that adds full modem FOTA support.
  See :ref:`SLM_AT_FOTA` for more information on how to use full modem FOTA functionality.

* :file:`overlay-cmux.conf` - Configuration file that adds support for the CMUX protocol.
  See :ref:`SLM_AT_CMUX` for more information.

* :file:`overlay-ppp.conf` - Configuration file that adds support for the Point-to-Point Protocol (PPP).

>   This disables most of the IP-based protocols available through AT commands (e.g. FTP, MQTT) as it is expected that the external MCU's own IP stack is used instead.

Reviewer: ```suggestion
  This disables most of the IP-based protocols available through AT commands (such as FTP and MQTT) as it is expected that the external MCU's own IP stack is used instead.
```

  See :ref:`CONFIG_SLM_PPP <CONFIG_SLM_PPP>` and :ref:`SLM_AT_PPP` for more information.

* :file:`overlay-ppp-without-cmux.overlay` - Devicetree overlay file that configures the UART to be used by PPP.
  This configuration file should be included when building SLM with PPP and without CMUX, in addition to :file:`overlay-ppp.conf`.

>   It can be customized to fit your configuration (UART, baud rate...).

Reviewer: ```suggestion
  It can be customized to fit your configuration (UART, baud rate, and so on).
```
